src/System.Private.ServiceModel/src/System/ServiceModel/Channels/TransportSecurityHelpers.cs

```diff
@@ -5,32 +5,199 @@
 using System.IdentityModel.Selectors;
 using System.IdentityModel.Tokens;
 using System.Net;
+using System.Net.Http;
+using System.Net.Security;
+using System.Runtime;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Principal;
 using System.ServiceModel.Security;
 using System.ServiceModel.Security.Tokens;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace System.ServiceModel.Channels
 {
     internal static class TransportSecurityHelpers
```

Comment:
Cool! it looks pretty close to my implementation which I was re-creating in a separate change :)